arviz/data/io_numpyro.py

```diff
@@ -410,14 +409,9 @@ def infer_dims(self) -> Dict[str, List[str]]:
 
     @requires("posterior")
     @requires("model")
-    @requires("coords")
     @requires("predictions")
     def infer_pred_dims(self) -> Dict[str, List[str]]:
-        if self.predictions_constant_data is not None:
-            dims = infer_dims(self.model, model_kwargs=self.predictions_constant_data)
```

Comment:
Assumption: removed this logic and chose to infer dims with the original input data in all cases - otherwise the `test_inference_data_constant_data` test was failing since it passed in `x2` as prediction constant data, even though x2 isnt a model input